notebooks/11-xarray-intro.md

```diff
@@ -1,185 +1,68 @@
-
-```{code-cell} ipython3
-ax = herstappe.plot.imshow(figsize=(12, 5))
-# ax.axes.set_aspect('equal')
-```
-
-```{code-cell} ipython3
-ax = herstappe.sel(band=1).plot.imshow(cmap="Reds", figsize=(12, 5))  #, robust=True)
-# ax.axes.set_aspect('equal')
-```
-
-As a preview, plot the intersection of the data at x coordinate closest to 400000 for each band:
+Let's start with reading the Sentinel RGB bands for Herstappe again:
 
 ```{code-cell} ipython3
-herstappe.sel(x=600_000, method='nearest').plot.line(col='band')
+herstappe_file = "./data/herstappe/raster/2020-09-17_Sentinel_2_L1C_True_color.tiff"
+herstappe = xr.open_rasterio(herstappe_file)
 ```
 
-But first, let's have a look at the data again:
-
 ```{code-cell} ipython3
 herstappe
 ```
 
-The output of xarray is a bit different to what we've previous seen. Let's go through the different elements:
-
-- It is a `xarray.DataArray`, one of the main data types provided by xarray
-- It has 3 __dimensions__:
-    - `band`: 3 bands (RGB)
-    - `y`: the y coordinates of the data set
-    - `x`: the x coordinates of the data set
-- Each of these dimensions are defined by a __coordinate__ (1D) array
-- Other metadata provided by the `tiff` are stored in the __`Attributes`__
-
-Looking to the data itself (click on the icons on the right), we can see this is still a Numpy array
-
-```{code-cell} ipython3
-herstappe.values
-```
-
-```{code-cell} ipython3
-type(herstappe.values)
-```
-
 ```{code-cell} ipython3
-herstappe.dtype
-```
-
-## Xarray adds context to NumPy arrays
-
-+++
-
-On of the most fundamental parts of the scientific python 'ecosystem' is [numpy](https://numpy.org/). A lot of other packages - you already used Pandas and GeoPandas in this course, and now also xarray - are built on top of NumPy and the `ndarray`  (n-dimensional array) object it provides.
-
-```{code-cell} ipython3
-import numpy as np
-```
-
-We could also read the GeoTiff data directly into a numpy array using `rasterio`:
-
-```{code-cell} ipython3
-import rasterio
-
-with rasterio.open(data_file) as src:
-    herstappe_array = src.read()
-    herstappe_meta = src.meta
-```
-
-```{code-cell} ipython3
-herstappe_array
-```
-
-As we learnt in the previous lesson, Rasterio returns a Numpy `ndarray`:
-
-```{code-cell} ipython3
-type(herstappe_array)
-```
-
-Numpy supports different `dtype`s (`float`, `int`,...), but all elements of an array do have the same dtype. Note that NumPy auto-detects the data-type from the input.
-
-```{code-cell} ipython3
-herstappe_array.dtype
-```
-
-The data type of this specific array `herstappe_array` is float32. More information on the data types Numpy supports is available in the [documentation](https://numpy.org/devdocs/user/basics.types.html#array-types-and-conversions-between-types).
-
-+++
-
-Converting to another data type is supported by `astype` method. When floats are preferred during calculation:
```

Comment:
We showed astype previously in the numpy section, but we still use it in the exercises. Maybe keep a mention of it?

Reply:
ok, I added a small section on it.